src/redux/history/methods.ts

```diff
@@ -5,56 +5,64 @@
  */
 
 import { getGlobalState, store } from '../index'
-import type {
-  HistoryEntry,
-  HistoryExportJson,
-  RemoveEntryAction,
-  SetEntriesAction,
-  UpdateEntryAction,
-  V1HistoryEntry
-} from './types'
-import { HistoryActionType, HistoryEntryOrigin } from './types'
+import type { HistoryExportJson, RemoveEntryAction, SetEntriesAction, UpdateEntryAction, V1HistoryEntry } from './types'
+import { HistoryActionType } from './types'
 import { download } from '../../components/common/download/download'
 import { DateTime } from 'luxon'
 import {
-  deleteHistory,
-  deleteHistoryEntry,
-  getHistory,
-  postHistory,
-  updateHistoryEntryPinStatus
+  deleteRemoteHistory,
+  deleteRemoteHistoryEntry,
+  getRemoteHistory,
+  setRemoteHistoryEntries,
+  updateRemoteHistoryEntryPinStatus
 } from '../../api/history'
-import {
-  historyEntryDtoToHistoryEntry,
-  historyEntryToHistoryEntryPutDto,
-  historyEntryToHistoryEntryUpdateDto
-} from '../../api/history/dto-methods'
+import { addRemoteOriginToHistoryEntry, historyEntryToHistoryEntryPutDto } from '../../api/history/dto-methods'
 import { Logger } from '../../utils/logger'
+import type { HistoryEntry, HistoryEntryWithOrigin } from '../../api/history/types'
+import { HistoryEntryOrigin } from '../../api/history/types'
 import { showErrorNotification } from '../ui-notifications/methods'
 
 const log = new Logger('Redux > History')
 
-export const setHistoryEntries = (entries: HistoryEntry[]): void => {
+/**
+ * Sets the given history entries into the current redux state and updates the local-storage.
+ * @param entries The history entries to set into the redux state.
+ */
+export const setHistoryEntries = (entries: HistoryEntryWithOrigin[]): void => {
   store.dispatch({
     type: HistoryActionType.SET_ENTRIES,
     entries
   } as SetEntriesAction)
   storeLocalHistory()
 }
 
-export const importHistoryEntries = (entries: HistoryEntry[]): Promise<void> => {
+/**
+ * Imports the given history entries into redux state and local-storage and remote based on their associated origin label.
+ * @param entries The history entries to import.
+ */
+export const importHistoryEntries = (entries: HistoryEntryWithOrigin[]): Promise<unknown> => {
+  // TODO Add filtering of entries by validating user-input
```

Comment:
Add issue

Reply:
Input validation is a general issue and not only scoped to this example.
I removed the todo, but we should not forget https://github.com/hedgedoc/react-client/issues/1219